pkg/agent/agent.go

```diff
@@ -305,7 +325,27 @@ func (c *agentController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 		return lastErr
 	}
 
-	return ctrl.Result{}, createOrUpdateMirrorSecrets()
+	if err := createOrUpdateMirrorSecrets(); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := c.createHostedClusterClaim(ctx, types.NamespacedName{Namespace: hc.Namespace, Name: hc.Status.KubeConfig.Name},
+		generateClusterClientFromSecret); err != nil {
+		if strings.Contains(err.Error(), "the server could not find the requested resource") ||
+			strings.Contains(err.Error(), "no such host") ||
```

Comment:
I am not sure if it is necessary to iterate all the error message. To keep it simple, maybe always requeue when there is an error